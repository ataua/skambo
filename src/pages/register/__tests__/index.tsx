import React from 'react'
import renderer from 'react-test-renderer'
import Register from '../index'
import 'jest-styled-components'

import 'mutationobserver-shim';

global.MutationObserver = window.MutationObserver;

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useHistory: jest.fn(),
  Link: () => <a></a>
}));

describe('Register', () => {
  it("Should render register", () => {
    const tree = renderer
      .create(<Register />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})