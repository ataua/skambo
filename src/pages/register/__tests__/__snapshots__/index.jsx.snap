// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Should render register 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 55%;
  margin: 15vh auto;
  width: 52%;
  height: 553px;
  justify-self: center;
  box-shadow: 0px 2px 6px rgba(0,0,0,0.25);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFF;
  width: 45%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 h1 {
  margin-bottom: 35px;
}

.c3 label {
  font-weight: 700;
  font-size: 14px;
  margin-bottom: 6px;
}

.c3 input {
  font-weight: 700;
  background-color: #F4F4F6;
  border: none;
  padding: 10px;
  border-radius: 4px;
  margin-bottom: 16px;
}

.c3 input::-webkit-input-placeholder {
  opacity: 64%;
}

.c3 input::-moz-placeholder {
  opacity: 64%;
}

.c3 input:-ms-input-placeholder {
  opacity: 64%;
}

.c3 input::placeholder {
  opacity: 64%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #8D70FB;
  width: 55%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  width: 100vw;
  height: 100vh;
  background-color: #fff;
}

.c0 a:hover {
  cursor: pointer;
}

.c5 {
  font-weight: 700;
  margin-top: 54px;
  color: #fff;
  background-color: #8D70FB;
  border-radius: 7px;
  height: 35px;
  width: 293px;
}

.c6 h3 {
  width: 287px;
  height: 45px;
  text-align: center;
  margin-top: 14px;
  font-size: 14px;
}

@media (max-width:1368px) {
  .c1 {
    width: 60%;
  }
}

@media (max-width:1200px) {
  .c1 {
    width: 75%;
  }
}

@media (max-width:992px) {
  .c1 {
    width: 93%;
    margin: 50px auto;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 96%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c3 {
    width: 96%;
  }

  .c3 h1 {
    margin-top: 30px;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 96%;
    margin-top: 64px;
  }
}

<div
  className="c0"
>
  <a
    onClick={[Function]}
  >
    Voltar
  </a>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        src="register-icon.svg"
      />
    </div>
    <div
      className="c3"
    >
      <h1>
        Cadastrar-se
      </h1>
      <form
        onSubmit={[Function]}
      >
        <div
          className="c4"
        >
          <label>
            E-mail
          </label>
          <input
            name="email"
            placeholder="Digite seu e-mail"
          />
        </div>
        <div
          className="c4"
        >
          <label>
            Senha
          </label>
          <input
            name="password"
            placeholder="Digite sua senha"
            type="password"
          />
        </div>
        <div>
          <button
            className="c5"
            type="submit"
          >
            Cadastrar!
          </button>
          <div
            className="c6"
          >
            <a
              onClick={[Function]}
            >
              <h3>
                 JÃ¡ possui conta? Entrar! 
              </h3>
            </a>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
